### PR TITLE
[ScheduledTask] Simplify check (+ fix typo in phpunit.xml)

### PR DESCRIPTION
**Pull request by apfelbox**
### 1. Why is this change necessary?
It makes the code simpler, easier to read and easier to maintain.
Also it uses the default tools of PHP a bit better.

### 2. What does this change do, exactly?
Instead of implementing a custom inheritance check, we just as PHP if one class extends another.

### 3. Describe each step to reproduce the issue or behaviour.
I was just reading the code and saw this small improvement.

### 4. Please link to the relevant issues (if any).
—

### 5. Checklist

- [ ] I have written tests and verified that they fail without my change
- [x] I have squashed any insignificant commits
- [x] I have created a [changelog file](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md) with all necessary information about my changes
- [ ] I have written or adjusted the documentation according to my changes
- [ ] This change has comments for package types, values, functions, and non-obvious lines of code
- [x] I have read the contribution requirements and fulfil them.

The existing tests should already cover this part.

### CODE REVIEW

**Comment by apfelbox** [comment]
Not sure if that Psalm issue is a false positive?

File: src/Core/Framework/MessageQueue/ScheduledTask/Scheduler/TaskScheduler.php [hunk 1]
```diff
@@ -121,9 +121,8 @@ private function buildCriteriaForAllScheduledTask(): Criteria
     private function queueTask(ScheduledTaskEntity $taskEntity): void
     {
         $taskClass = $taskEntity->getScheduledTaskClass();
-        $taskClassParents = class_parents($taskClass);
 
-        if ($taskClassParents === false || !\in_array(ScheduledTask::class, $taskClassParents, true)) {
+        if (!\is_a($taskClass, ScheduledTask::class, true)) {
```
Comment (mitelg) [inline src/Core/Framework/MessageQueue/ScheduledTask/Scheduler/TaskScheduler.php:125]: due to this change, Psalm can now determine which object type the `$task` variable will have. 

that is why the annotation from line 132 is unnecessary

Comment (apfelbox) [reply src/Core/Framework/MessageQueue/ScheduledTask/Scheduler/TaskScheduler.php:125]: Oh wow, didn’t expect that level of smartness 😅. Will change 👍

**Comment by J-Rahe** [comment]
Hey : )
We've just set your PR to “Scheduled”.
This means our initial screening is passed.
Your PR will now be reviewed and tested before being put trough our internal extended testsuite and review process.
We will keep you informed wether something has to be done by you or when the PR is merged : )
If you have any questions feel free to ask.

**Comment by J-Rahe** [comment]
Hey @apfelbox, i missed that before, but could you please add an [changelog](https://github.com/shopware/platform/blob/trunk/adr/2020-08-03-Implement-New-Changelog.md)?
We cant merge this without one.

**Comment by apfelbox** [comment]
@J-Rahe no worries, just pushed a changelog entry.

I wasn't sure whether I should add entry even for such a small change, but now it is included. Please send me a quick note, if something is not all right 👍

**Comment by J-Rahe** [comment]
Hey @apfelbox, your PR just got merged : )
I did some small changes to the changelog to use our [keywords](https://jira.shopware.com/browse/NEXT-11691).

Thanks :+1:

**Pull request closed without merging**